.clang-tidy

```diff
@@ -24,8 +24,29 @@ Checks: >
   -bugprone-switch-missing-default-case,
   -bugprone-unsafe-functions,
```

Comment:
does this have many false positives? if not , does clang-tidy have a way to let us ignore the FPs, then we could enable this?

Reply:
Two warnings

```
/Users/dundargoc/programs/neovim/src/nvim/main.c:350:3: error: function 'setbuf' has no error detection; 'setvbuf' should be used instead [bugprone-unsafe-functions,-warnings-as-errors]
  350 |   setbuf(stdout, NULL);
      |   ^~~~~~
```

and

```
/Users/dundargoc/programs/neovim/src/nvim/tag.c:1562:7: error: function 'rewind' has no error detection; 'fseek' should be used instead [bugprone-unsafe-functions,-warnings-as-errors]
 1562 |       rewind(st->fp);
      |       ^~~~~~
```

Reply:
Nah, no way to ignore specific cases that I could find. We'll have to rely on clint for now if we want this I think.

Reply:
Alternatively we can disable the setbuf warning and hope that we don't run into false positives too often?

Reply:
Sure, or maybe we can avoid setbuf / rewind.